api/src/test/java/org/apache/iceberg/transforms/TestTimestamps.java

```diff
@@ -241,4 +243,54 @@ public void testTimestampsReturnType() {
     Type hourResultType = hour.getResultType(type);
     assertThat(hourResultType).isEqualTo(Types.IntegerType.get());
   }
+
+  @Test
+  public void apply_bad_source_type() {
+    Timestamps badSourceType =
+        new Timestamps(ChronoUnit.CENTURIES, Timestamps.ResultTypeUnit.YEARS, "year");
+    assertThatThrownBy(() -> badSourceType.apply(11L))
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessageMatching("Unsupported source type unit: Centuries");
+  }
+
+  @Test
+  public void apply_bad_result_type() {
```

Comment:
```suggestion
  public void applyBadResultType() {
```

Reply:
please also update the other test method names accordingly, as the codebase typically doesn't use `_` in method names

Reply:
@nastra just curious, can this be checked statically, perhaps with a tool like checkstyle?

Reply:
done